### DISCUSSION TITLE
Whisper won't fully transcribe the mp3 file

### DISCUSSION THREAD

**Discussion by Potatobunnana** [openai/whisper#442] [Q&A]
Hi,

I am transcribing mp3 files and I noticed that mp3s with a long duration (around 20 mins) doesn't transcribe for the full duration of the mp3 file.

For instance, I was transcribing a mp3 file from yt (https://www.youtube.com/watch?v=ALcvPC-yDLs) and ran the following code:

import whisper

model = whisper.load_model("base.en")
mp3 = 'ALcvPC-yDLs.mp3'
result = model.transcribe(mp3, language = "en", fp16 = False)
print(result["text"])

and my text output ends early.

What's odd is I ran this exact code hours ago and got the full extraction but I tried to use GPU processing instead of CPU. I thought this was the issue, as it was the only variable I changed so I reinstalled everything on a different laptop but found that I still didn't get the full transcription of the mp3.  

Any help would be greatly appreciated.

Thank You!

**Comment by srujohn** [comment]
Hello,
Have you found any solution for this error???\

**Reply by turnkit** [reply]
What happens if you transcode to wav first then run whisper in the wav files?

I had this same problem early on and switched to my wave versions and the issue seemed to clear up.

**Reply by Potatobunnana** [reply]
What I found was there must be a limit to how much you can print a variable into the terminal.

I printed the variable into a word docx instead and found the whole transcript.

**Reply by MrEdwards007** [reply]
That could be the case but I have printed to screen a little more than an hour of continuous text from an audio book.  My only challenge was ensuring that it would actually print, adding utf-8 encoding, e.g. print(allText.encode('utf-8'))

**Reply by srujohn** [reply]
> What I found was there must be a limit to how much you can print a variable into the terminal.
> 
> I printed the variable into a word docx instead and found the whole transcript.

I have tried but still not working, will you help me with code. below is my code.

result = whisper.decode(model, mel, options)

print(result.text)
print(len(result.text))

with open('result.txt', 'w') as f:
     f.write(result.text)

**Comment by FurkanGozukara** [comment]
i use command line it does transcribe even 2 hours of mp3 file

actually i provide mp3 file instead of native mkv since it performs better

you can watch my tutorial videos

https://www.youtube.com/watch?v=msj3wuYf3d8

https://www.youtube.com/watch?v=dP53wzLwqMA

https://www.youtube.com/watch?v=q8Q8CCDdSKo
